### PR TITLE
fix: [CCM-7340]: GCP connector project id issue

### PR DESCRIPTION
**Pull request by jyotiarora-harness**
##### Summary:

For CE GCP connector project id was not getting saved while editing

##### Jira Links:

https://harness.atlassian.net/browse/CCM-7340

##### Screenshots:

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Comment by jyotiarora-harness** [comment]
retrigger titlecheck

**Comment by jyotiarora-harness** [comment]
retrigger titlecheck

**Review by rajarshee-harness — APPROVED** [review]

File: src/modules/35-connectors/components/CreateConnector/CEGcpConnector/steps/ChooseRequirements.tsx [hunk 1]
```diff
@@ -96,10 +96,10 @@ const ChooseRequirements: React.FC<StepProps<CEGcpConnectorDTO>> = ({ prevStepDa
       features.push('BILLING')
     }
     const newspec: GcpCloudCostConnector = {
+      projectId: '',
       ...prevStepData?.spec,
       featuresEnabled: features,
-      serviceAccountEmail: defaultTo(prevStepData?.serviceAccount, ''),
-      projectId: ''
```
Comment (jyotiarora-harness) [inline src/modules/35-connectors/components/CreateConnector/CEGcpConnector/steps/ChooseRequirements.tsx:102]: Changed the sequence of projectId as the projectId is coming in `prevStepData.spec` and this line was replacing that value

Comment (jyotiarora-harness) [inline src/modules/35-connectors/components/CreateConnector/CEGcpConnector/steps/ChooseRequirements.tsx:102]: Changed the sequence of projectId as the projectId is coming in `prevStepData.spec` and this line was replacing that value

**Review by rishik30 — APPROVED** [review]

**Review by avinashmadhwani02 — APPROVED** [review]

**Pull request merged**